sources/System/MenuControllable/MenuControllableSystem.cpp

```diff
@@ -45,21 +51,64 @@ namespace BBM
 	}
 
-	void MenuControllableSystem::onFixedUpdate(WAL::ViewEntity<ControllableComponent> &entity)
+	bool MenuControllableSystem::_mouseOnButton(WAL::ViewEntity<OnClickComponent, OnHoverComponent, OnIdleComponent, PositionComponent, Drawable2DComponent> &entity) const
+	{
+		auto &positionComponent = entity.get<PositionComponent>();
+		RAY::Vector2 rayMousePos = RAYControl::Mouse::getCursorPosition();
+		RAY::Texture *texture = dynamic_cast<RAY::Texture *>(entity.get<Drawable2DComponent>().drawable.get());
+		RAY2D::Text *text = dynamic_cast<RAY2D::Text *>(entity.get<Drawable2DComponent>().drawable.get());
+		Vector2f buttonPos(positionComponent.getX(), positionComponent.getY());
+		Vector2f mousePos(rayMousePos.x, rayMousePos.y);
+		Vector2f dimensions;
+
+		WAL::Entity *newButton = nullptr; 
+
+		if (texture) {
+			dimensions.x = texture->getDimensions().x;
+			dimensions.y = texture->getDimensions().y;
+		} else if (text) {
+			dimensions.y = text->getFontSize();
+			dimensions.x = text->getString().size() * (text->getLetterSpacing() + text->getFontSize());
+		} else
+			return false;
+		if ((buttonPos.x <= mousePos.x && mousePos.x <= buttonPos.x + dimensions.x)
+		&& (buttonPos.y <= mousePos.y && mousePos.y <= buttonPos.y + dimensions.y))
+			return true;
+		return false;
+	}
+
+	void MenuControllableSystem::onSelfUpdate()
 	{
-		auto &controllable = entity.get<ControllableComponent>();
-		auto &buttons = _wal.getScene()->view<OnClickComponent, OnHoverComponent, OnIdleComponent>();
+
+		auto &controllableView = this->_wal.getScene()->view<ControllableComponent>();
+		auto &buttons = _wal.getScene()->view<OnClickComponent, OnHoverComponent, OnIdleComponent, PositionComponent, Drawable2DComponent>();
+
 
 		if (this->_currentButton && this->_currentButton->_scene.getID() != this->_wal.getScene()->getID()) {
 			this->_currentButton->getComponent<OnIdleComponent>().onEvent(*this->_currentButton, this->_wal);
 			this->_currentButton = nullptr;
+			return;
 		}
 		if (this->_currentButton == nullptr && buttons.size()) {
 			this->_currentButton = &(*buttons.front());
 			this->_currentButton->getComponent<OnHoverComponent>().onEvent(*this->_currentButton, this->_wal);
 		}
 		if (!this->_currentButton)
 			return;
-		this->_updateCurrentButton(controllable.jump, controllable.move);
+		for (auto &[_, controllable]: controllableView)
+			if (controllable.move.x || controllable.move.y || controllable.jump) {
+				this->_updateCurrentButton(controllable.jump, controllable.move);
+				return;
+			}
+		for (auto &entity:  buttons) {
+			if (_mouseOnButton(entity)) {
+				if (this->_currentButton)
+					this->_currentButton->getComponent<OnIdleComponent>().onEvent(*this->_currentButton, this->_wal);
+				this->_currentButton = &(*entity);
+				this->_currentButton->getComponent<OnHoverComponent>().onEvent(*this->_currentButton, this->_wal);
+				if (RAYControl::Mouse::isPressed(RAYControl::Mouse::Button::MOUSE_BUTTON_LEFT))
```

Comment:
On the controller click, there is a delay set (one can't press a button more than once every 150ms. Are you sure you don't need a similar limitation with the mouse?

Reply:
No, since I'm using isPressed, instead of isDown